ipa-core/src/protocol/hybrid/shuffle.rs

```diff
@@ -0,0 +1,248 @@
+use std::ops::Add;
+
+use generic_array::{ArrayLength, GenericArray};
+use typenum::{Unsigned, U3, U8};
+
+use crate::{
+    error::{Error, UnwrapInfallible},
+    ff::{boolean_array::BA64, Serializable},
+    protocol::{
+        ipa_prf::shuffle::sharded::Shuffleable,
+        prss::{FromRandom, FromRandomU128},
+    },
+    report::hybrid::IndistinguishableHybridReport,
+    secret_sharing::{replicated::ReplicatedSecretSharing, Sendable, SharedValue},
+};
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct IndistinguishableHybridShare<BK, V> {
+    match_key: BA64,
+    value: V,
+    breakdown_key: BK,
+}
+
+impl<BK, V> Serializable for IndistinguishableHybridShare<BK, V>
+where
+    BK: SharedValue,
+    V: SharedValue,
+    <BK as Serializable>::Size: Add<U8>,
```

Comment:
It was really unclear why this needs to be a `U8`, and this doesn't work with `Add<<BA64 as Serializable>::Size>` which seems to resolve to `U8`. Very open to any suggestions to how we may clean this up.